sycl/plugins/opencl/pi_opencl.cpp

```diff
@@ -454,6 +492,322 @@ pi_result OCL(piEnqueueMemBufferMap)(
   return ret_err;
 }
 
+//
+// USM
+//
+
+// Allocates host memory accessible by the device.
+pi_result OCL(piHostMemAlloc)(void **result_ptr, pi_context context,
+                              pi_usm_mem_properties *properties, size_t size,
+                              pi_uint32 alignment) {
+
+  void *Ptr = nullptr;
+  pi_result RetVal = PI_INVALID_OPERATION;
+
+  // First we need to look up the function pointer
+  // It would be good if we could store it for future reuse
+  // Are statics ok?
+  clHostMemAllocINTEL_fn FuncPtr;
+  RetVal = getExtFuncFromContext<clHostMemAllocINTEL_fn>(
```

Comment:
This will repeat discovery of the needed function in RT for every PI call, which is clearly an overkill. We need to find a way to cache it.

Reply:
>  Maybe the found function pointer should just replace the pointer to "piHostMemAlloc" interface in the pi_plugin table of interfaces, so whoever calls piHostMemAlloc again will get directly into "clHostMemAllocINTEL"

I realized that this will not work since the OpenCL interface is different from PI interface

Reply:
static thread_local?

Reply:
Just pushed a new commit that seems to handle this in a thread safe way.